### ISSUE TITLE
Workspaces support for polling

### ISSUE THREAD

**Issue by izaaklauer** [hashicorp/waypoint#2644]
Polling (both git and app polling) is currently only supported in the default workspace.

**Comment by izaaklauer** [comment]
Currently blocked by https://github.com/hashicorp/waypoint/issues/1665 - we need some way to configure polling differently per-workspace.

**Comment by denysvitali** [comment]
Any update here? #1665 is also open but a comment in #1815 says:
> Waypoint has added and changed a lot around workspaces since it was opened. Are you willing to have another look at what this does and what waypoint does and see if it's still missing functionality for you?

Sadly the polling seems to be hardcoded to `default` and thus it's not possible to use a monorepo approach with three branches like our model:
- `develop`
- `staging`
- `production`

The issue seems here, already noted as a TODO by @mitchellh:
https://github.com/hashicorp/waypoint/blob/96d7c7c279ee513617964577415fed1cacae7553/internal/runner/operation_poll.go#L38-L51

We would love to see this feature implemented / know if we have to implement it ourselves again, so that we can consider Waypoint mature enough to be used in our environment. Sadly the version of waypoint we're using diverged from the upstream one, and keeping them in sync has become too much of a burden - we can definitely merge the efforts and provide a solution that can be upstreamed :D

cc/ @krzysiekb

**Comment by krzysiekb** [comment]
Hi @briancain - Thanks for the info. For now, we implemented a workaround using external tool for polling, but it would be nice to have a waypoint native solution :-)
